src/window.rs

```diff
@@ -710,11 +706,29 @@ pub trait Window: AsAny + Send + Sync {
     ///
     /// ## Platform-specific
     ///
-    /// - **iOS:** Returns the [`PhysicalSize`] of the window in screen space coordinates.
     /// - **Web:** Returns the size of the canvas element. _Note: this returns the same value as
     ///   [`Window::surface_size`]._
     fn outer_size(&self) -> PhysicalSize<u32>;
 
+    /// The area of the surface that is unobstructed.
+    ///
+    /// On some devices, especially mobile devices, the screen is not a perfect rectangle, and may
+    /// have rounded corners, notches, bezels, and so on. When drawing your content, you usually
+    /// want to draw your background and other such unimportant content on the entire surface, while
+    /// you will want to restrict important content such as text, interactable or visual indicators
+    /// to the part of the screen that is actually visible; for this, you use the safe area.
+    ///
+    /// The safe area is a rectangle that is defined relative to the origin at the top-left corner
+    /// of the surface, and the size extending downwards to the right. The area will not extend
+    /// beyond [the bounds of the surface][Window::surface_size].
+    ///
+    /// ## Platform-specific
+    ///
+    /// - **Wayland / Android / Orbital:** Unimplemented, returns `((0, 0), surface_size)`.
+    /// - **macOS:** This must be used when using `set_simple_fullscreen` to prevent overlapping the
+    ///   notch on newer devices.
+    fn safe_area(&self) -> (PhysicalPosition<u32>, PhysicalSize<u32>);
```

Comment:
You can default implement it with `surface_position` and `surface_size`.

Reply:
Yeah, this would work:

```suggestion
    fn safe_area(&self) -> (PhysicalPosition<u32>, PhysicalSize<u32>) {
        ((0, 0).into(), self.surface_size())
    }
```

But not sure if it's a good idea, it might make backends forget to implement it? Idk., we haven't really figured out the story for when to default impl things on `Window`, and there's a lot of other places it could be done, so I'd rather avoid doing so in this PR, and instead discuss it separately.

Reply:
In this particular case, the API is not that wide spread, thus likely won't be implemented by many backends.

Reply:
I would like to keep things in this PR consistent with what we're already doing in the codebase. If we want to change how it's done, then it should go in another PR, because it applies to a lot of properties (`set_min_surface_size`, `surface_resize_increments`, `is_minimized` etc.), not just `safe_area`.

Reply:
I'm in favor of **not** default implementing anything.